artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ByteUtil.java

```diff
@@ -53,19 +53,27 @@
 
    public static void debugFrame(Logger logger, String message, ByteBuf byteIn) {
       if (logger.isTraceEnabled()) {
-         int location = byteIn.readerIndex();
-         // debugging
-         byte[] frame = new byte[byteIn.writerIndex()];
-         byteIn.readBytes(frame);
+         outFrame(logger, message, byteIn, false);
+      }
+   }
 
-         try {
+   public static void outFrame(Logger logger, String message, ByteBuf byteIn, boolean info) {
+      int location = byteIn.readerIndex();
+      // debugging
```

Comment:
This can became a proper debug/trace info

Reply:
actually... this one IMO should still be System.out.println(..)

it's a debug method, right?